x/mint/keeper/keeper.go

```diff
@@ -137,56 +143,99 @@ func (k Keeper) GetProportions(ctx sdk.Context, mintedCoin sdk.Coin, ratio sdk.D
-	err = k.bankKeeper.BurnCoins(ctx, types.ModuleName, participationRewardCoins)
+	// allocate pool allocation ratio to community growth pool
+	communityPoolGrowthAddress := k.GetSubmoduleAddress(types.CommunityGrowthSubmoduleName)
+	communityPoolGrowthProportion := k.GetProportions(ctx, mintedCoin, proportions.CommunityPoolGrowth)
+	communityPoolGrowthCoins := sdk.NewCoins(communityPoolGrowthProportion)
+	k.Logger(ctx).Info(fmt.Sprintf("\t\t\t...community growth: %d to %s", communityPoolGrowthProportion.Amount.Int64(), communityPoolGrowthAddress))
+	err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, communityPoolGrowthAddress, communityPoolGrowthCoins)
 	if err != nil {
 		return err
 	}
 
-	// Take the current balance of the developer rewards pool and remove it from the supply offset
-	// We re-introduce the new supply at the end, in order to avoid any rounding discrepancies.
-	// developerAccountBalance := k.bankKeeper.GetBalance(ctx, k.accountKeeper.GetModuleAddress(types.DeveloperVestingModuleAcctName), mintedCoin.Denom)
-	// k.bankKeeper.AddSupplyOffset(ctx, mintedCoin.Denom, developerAccountBalance.Amount)
-
-	// // Take the new balance of the developer rewards pool and add it back to the supply offset deduction
-	// developerAccountBalance = k.bankKeeper.GetBalance(ctx, k.accountKeeper.GetModuleAddress(types.DeveloperVestingModuleAcctName), mintedCoin.Denom)
-	// k.bankKeeper.AddSupplyOffset(ctx, mintedCoin.Denom, developerAccountBalance.Amount.Neg())
+	// allocate pool allocation ratio to security budget pool
+	communityPoolSecurityBudgetAddress := k.GetSubmoduleAddress(types.CommunitySecurityBudgetSubmoduleName)
+	communityPoolSecurityBudgetProportion := k.GetProportions(ctx, mintedCoin, proportions.CommunityPoolSecurityBudget)
+	communityPoolSecurityBudgetCoins := sdk.NewCoins(communityPoolSecurityBudgetProportion)
+	k.Logger(ctx).Info(fmt.Sprintf("\t\t\t...community security budget: %d to %s", communityPoolSecurityBudgetProportion.Amount.Int64(), communityPoolSecurityBudgetAddress))
+	err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, communityPoolSecurityBudgetAddress, communityPoolSecurityBudgetCoins)
+	if err != nil {
+		return err
+	}
 
-	// subtract from original provision to ensure no coins left over after the allocations
-	communityPoolCoins := sdk.NewCoins(mintedCoin).Sub(stakingIncentivesCoins).Sub(poolIncentivesCoins).Sub(participationRewardCoins)
-	err = k.distrKeeper.FundCommunityPool(ctx, communityPoolCoins, k.accountKeeper.GetModuleAddress(types.ModuleName))
+	// sweep any remaining tokens to the community growth pool (this should NEVER happen, barring rounding imprecision)
+	remainingCoins := sdk.NewCoins(mintedCoin).
+		Sub(stakingIncentivesCoins).
+		Sub(strategicReserveCoins).
+		Sub(communityPoolGrowthCoins).
+		Sub(communityPoolSecurityBudgetCoins)
+	err = k.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, communityPoolGrowthAddress, remainingCoins)
```

Comment:
Is it worth pushing the `SendCoinsFromModuleTo...` calls to the bottom of the function, and checking that the coins sum up to `mintedCoin` before distributing?

Reply:
Good idea, done

Reply:
added a check that the remaining is LT 5% of the total. there is rounding imprecision, so a 0 check breaks